### PR TITLE
missing memcachedExporter resources + imagePullSecrets

### LINKED ISSUE #2320: [HELM][Memcached] missing resources memcachedExporter + imagePullSecrets
#### Is your feature request related to a problem? Please describe.
- On our kubernetes clusters we don't allow pod to be scheduled is there is no resources defined
Right now I'm stuck using the Memcached component because the manifest doesn't allow me to set resources for memcachedExporter.
- We pull images from our private Artifactory don't bypass the limitation of pulls from docker.io

#### Describe the solution you'd like
I'm opening a pr to improve those 2 missing parts 👍 

#### Describe alternatives you've considered
n/a

#### Additional context
n/a

### PR DESCRIPTION
**Pull request by micborens**
#### What this PR does
- availability to pull image from private registry (like artifactory)
- missing resources block for memcachedExporter

#### Which issue(s) this PR fixes or relates to

Fixes #2320 

#### Checklist

- [x] Tests updated
- [n/a] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: operations/helm/charts/mimir-distributed/CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,9 @@ Entries should be ordered as follows:
 Entries should include a reference to the Pull Request that introduced the change.
 
 ## main / unreleased
+* [ENHANCEMENT] Update memcached statefulset manifest #2321
```
Comment (krajorama) [inline operations/helm/charts/mimir-distributed/CHANGELOG.md:15]: please move  to top of the other ENHACEMENTs

**Review by krajorama — COMMENTED** [review]
Thank you for the contribution, I've made some suggestions

File: operations/helm/charts/mimir-distributed/templates/memcached/_memcached-statefulset.tpl [hunk 2]
```diff
@@ -54,6 +54,12 @@ spec:
       tolerations:
         {{- toYaml .tolerations | nindent 8 }}
       terminationGracePeriodSeconds: {{ .terminationGracePeriodSeconds }}
+      {{- if $.ctx.Values.memcached.image.pullSecrets }}
```
[the remark this answers is not in the document]
Comment (micborens) [reply operations/helm/charts/mimir-distributed/templates/memcached/_memcached-statefulset.tpl:57]: Sure i will update to reuse the existing value

File: operations/helm/charts/mimir-distributed/CHANGELOG.md (hunk 1 above)
Comment (micborens) [reply operations/helm/charts/mimir-distributed/CHANGELOG.md:15]: you mean 1 enhancement per bullet point?

File: operations/helm/charts/mimir-distributed/templates/memcached/_memcached-statefulset.tpl (hunk 2 above)
[the remark this answers is not in the document]
Comment (micborens) [reply operations/helm/charts/mimir-distributed/templates/memcached/_memcached-statefulset.tpl]: I've re-used the secrets from the global.

File: operations/helm/charts/mimir-distributed/values.yaml [hunk 3]
```diff
@@ -1051,6 +1052,13 @@ memcachedExporter:
     tag: v0.6.0
     pullPolicy: IfNotPresent
 
+  resources:
+    requests:
+      memory: "256Mi"
+      cpu: "200m"
+    limits:
+      memory: "256Mi"
```
[the remark this answers is not in the document]
Comment (micborens) [reply operations/helm/charts/mimir-distributed/values.yaml]: I've removed the limitation to respect this recommendation.

File: operations/helm/charts/mimir-distributed/CHANGELOG.md (hunk 1 above)
Comment (micborens) [reply operations/helm/charts/mimir-distributed/CHANGELOG.md:15]: ah no ok got it. nvm. i push the update

**Comment by krajorama** [comment]
Do you have the ability to run `make check-helm-tests` in the repository root in your environment? The golden record of kube manifests needs an update, see failed `ci / lint-helm`.

**Comment by micborens** [comment]
I've tested and run the ` make check-helm-tests` command.
I've commit tests updated. i hope ci will pass with success 🤞

**Comment by micborens** [comment]
FYI, i've rebase my branch from `grafana/mimir:main` branch. 
But now `make check-helm-tests` return an error:
```Templating test-enterprise-configmap-values
Error: template: mimir-distributed/templates/store-gateway/store-gateway-statefulset.yaml:48:12: executing "mimir-distributed/templates/store-gateway/store-gateway-statefulset.yaml" at <include "mimir.podAnnotations" (dict "ctx" . "component" "store-gateway")>: error calling include: template: mimir-distributed/templates/_helpers.tpl:245:20: executing "mimir.podAnnotations" at <include (print .ctx.Template.BasePath "/mimir-config.yaml") .ctx>: error calling include: template: mimir-distributed/templates/mimir-config.yaml:19:7: executing "mimir-distributed/templates/mimir-config.yaml" at <include "mimir.calculatedConfig" .>: error calling include: template: mimir-distributed/templates/_helpers.tpl:99:24: executing "mimir.calculatedConfig" at <include (print $.Template.BasePath "/_config-render.tpl") .>: error calling include: template: mimir-distributed/templates/_config-render.tpl:1:4: executing "mimir-distributed/templates/_config-render.tpl" at <tpl .Values.mimir.config .>: error calling tpl: error during tpl function execution for "activity_tracker:\n  filepath: /data/metrics-activity.log\n\n{{- if .Values.enterprise.enabled }}\nadmin_api:\n  leader_election:\n    enabled: true\n    ring:\n      kvstore:\n        store: \"memberlist\"\n\n  {{- if .Values.minio.enabled }}\nadmin_client:\n  storage:\n    type: s3\n    s3:\n      access_key_id: {{ .Values.minio.accessKey }}\n      bucket_name: enterprise-metrics-admin\n      endpoint: {{ .Release.Name }}-minio.{{ .Release.Namespace }}.svc:9000\n      insecure: true\n      secret_access_key: {{ .Values.minio.secretKey }}\n  {{- end }}\n{{- end }}\n\nalertmanager:\n  data_dir: /data\n  enable_api: true\n  external_url: /alertmanager\n\n{{- if .Values.minio.enabled }}\nalertmanager_storage:\n  backend: s3\n  s3:\n    access_key_id: {{ .Values.minio.accessKey }}\n    bucket_name: {{ include \"mimir.minioBucketPrefix\" . }}-ruler\n    endpoint: {{ .Release.Name }}-minio.{{ .Release.Namespace }}.svc:9000\n    insecure: true\n    secret_access_key: {{ .Values.minio.secretKey }}\n{{- end }}\n\n{{- if .Values.enterprise.enabled }}\nauth:\n  type: enterprise\n{{- end }}\n\n# This configures how the store-gateway synchronizes blocks stored in the bucket. It uses Minio by default for getting started (configured via flags) but this should be changed for production deployments.\nblocks_storage:\n  backend: s3\n  bucket_store:\n    {{- if index .Values \"chunks-cache\" \"enabled\" }}\n    chunks_cache:\n      backend: memcached\n      memcached:\n        addresses: {{ include \"mimir.chunksCacheAddress\" . }}\n        max_item_size: {{ mul (index .Values \"chunks-cache\").maxItemMemory 1024 1024 }}\n        timeout: 450ms\n    {{- end }}\n    {{- if index .Values \"index-cache\" \"enabled\" }}\n    index_cache:\n      backend: memcached\n      memcached:\n        addresses: {{ include \"mimir.indexCacheAddress\" . }}\n        max_item_size: {{ mul (index .Values \"index-cache\").maxItemMemory 1024 1024 }}\n    {{- end }}\n    {{- if index .Values \"metadata-cache\" \"enabled\" }}\n    metadata_cache:\n      backend: memcached\n      memcached:\n        addresses: {{ include \"mimir.metadataCacheAddress\" . }}\n        max_item_size: {{ mul (index .Values \"metadata-cache\").maxItemMemory 1024 1024 }}\n    {{- end }}\n    sync_dir: /data/tsdb-sync\n  {{- if .Values.minio.enabled }}\n  s3:\n    access_key_id: {{ .Values.minio.accessKey }}\n    bucket_name: {{ include \"mimir.minioBucketPrefix\" . }}-tsdb\n    endpoint: {{ .Release.Name }}-minio.{{ .Release.Namespace }}.svc:9000\n    insecure: true\n    secret_access_key: {{ .Values.minio.secretKey }}\n  {{- end }}\n  tsdb:\n    dir: /data/tsdb\n\n{{- if .Values.enterprise.enabled }}\ncluster_name: \"{{ .Release.Name }}\"\n{{- end }}\n\ncompactor:\n  data_dir: \"/data\"\n\nfrontend:\n  align_queries_with_step: true\n  log_queries_longer_than: 10s\n  {{- if index .Values \"results-cache\" \"enabled\" }}\n  results_cache:\n    backend: memcached\n    memcached:\n      addresses: {{ include \"mimir.resultsCacheAddress\" . }}\n      max_item_size: {{ mul (index .Values \"results-cache\").maxItemMemory 1024 1024 }}\n  cache_results: true\n  {{- end }}\n\nfrontend_worker:\n  frontend_address: {{ template \"mimir.fullname\" . }}-query-frontend-headless.{{ .Release.Namespace }}.svc:{{ include \"mimir.serverGrpcListenPort\" . }}\n\n{{- if and .Values.enterprise.enabled .Values.gateway.useDefaultProxyURLs }}\ngateway:\n  proxy:\n    admin_api:\n      url: http://{{ template \"mimir.fullname\" . }}-admin-api.{{ .Release.Namespace }}.svc:{{ include \"mimir.serverHttpListenPort\" . }}\n    alertmanager:\n      url: http://{{ template \"mimir.fullname\" . }}-alertmanager.{{ .Release.Namespace }}.svc:{{ include \"mimir.serverHttpListenPort\" . }}\n    compactor:\n      url: http://{{ template \"mimir.fullname\" . }}-compactor.{{ .Release.Namespace }}.svc:{{ include \"mimir.serverHttpListenPort\" . }}\n    default:\n      url: http://{{ template \"mimir.fullname\" . }}-admin-api.{{ .Release.Namespace }}.svc:{{ include \"mimir.serverHttpListenPort\" . }}\n    distributor:\n      url: dns:///{{ template \"mimir.fullname\" . }}-distributor-headless.{{ .Release.Namespace }}.svc.{{ .Values.global.clusterDomain }}:{{ include \"mimir.serverGrpcListenPort\" . }}\n    ingester:\n      url: http://{{ template \"mimir.fullname\" . }}-ingester.{{ .Release.Namespace }}.svc:{{ include \"mimir.serverHttpListenPort\" . }}\n    query_frontend:\n      url: http://{{ template \"mimir.fullname\" . }}-query-frontend.{{ .Release.Namespace }}.svc:{{ include \"mimir.serverHttpListenPort\" . }}\n    ruler:\n      url: http://{{ template \"mimir.fullname\" . }}-ruler.{{ .Release.Namespace }}.svc:{{ include \"mimir.serverHttpListenPort\" . }}\n    store_gateway:\n      url: http://{{ template \"mimir.fullname\" . }}-store-gateway.{{ .Release.Namespace }}.svc:{{ include \"mimir.serverHttpListenPort\" . }}\n{{- end }}\n\ningester:\n  ring:\n    final_sleep: 0s\n    num_tokens: 512\n    unregister_on_shutdown: false\n\ningester_client:\n  grpc_client_config:\n    max_recv_msg_size: 104857600\n    max_send_msg_size: 104857600\n\n{{- if .Values.enterprise.enabled }}\ninstrumentation:\n  enabled: true\n  distributor_client:\n    address: dns:///{{ template \"mimir.fullname\" . }}-distributor-headless.{{ .Release.Namespace }}.svc.{{ .Values.global.clusterDomain }}:{{ include \"mimir.serverGrpcListenPort\" . }}\n\nlicense:\n  path: \"/license/license.jwt\"\n{{- end }}\n\nlimits: {}\n\nmemberlist:\n  abort_if_cluster_join_fails: false\n  compression_enabled: false\n  join_members:\n  - {{ include \"mimir.fullname\" . }}-gossip-ring\n\nruler:\n  alertmanager_url: dnssrvnoa+http://_http-metrics._tcp.{{ template \"mimir.fullname\" . }}-alertmanager-headless.{{ .Release.Namespace }}.svc.{{ .Values.global.clusterDomain }}/alertmanager\n  enable_api: true\n  rule_path: /data\n\n{{- if .Values.minio.enabled }}\nruler_storage:\n  backend: s3\n  s3:\n    endpoint: {{ .Release.Name }}-minio.{{ .Release.Namespace }}.svc:9000\n    bucket_name: {{ include \"mimir.minioBucketPrefix\" . }}-ruler\n    access_key_id: {{ .Values.minio.accessKey }}\n    secret_access_key: {{ .Values.minio.secretKey }}\n    insecure: true\n{{- end }}\n\nruntime_config:\n  file: /var/{{ include \"mimir.name\" . }}/runtime.yaml\n\nserver:\n  grpc_server_max_concurrent_streams: 1000\n": template: mimir-distributed/templates/store-gateway/store-gateway-statefulset.yaml:50:11: executing "mimir-distributed/templates/store-gateway/store-gateway-statefulset.yaml" at <index .Values "chunks-cache" "enabled">: error calling index: index of nil pointer```

**Comment by micborens** [comment]
@krajorama is it mergeable for you ? 🙏

**Review by krajorama — APPROVED** [review]
LGTM

**Comment by osg-grafana** [comment]
This closure was unintentional.

**Pull request merged**